adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -378,132 +369,131 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
       ContextUtil.getConfiguration(job))
     if (Metrics.isRecording) records.instrument() else records
 
-    applyPredicate(records.map(p => vcc.convertToAnnotation(p._2.get)), predicate)
+    records.map(p => vcc.convertToAnnotation(p._2.get))
   }
 
-  def loadParquetVariantAnnotations[U <: ADAMPredicate[DatabaseVariantAnnotation]](
+  def loadParquetVariantAnnotations(
     filePath: String,
-    predicate: Option[Class[U]] = None,
+    predicate: Option[FilterPredicate] = None,
     projection: Option[Schema] = None): RDD[DatabaseVariantAnnotation] = {
-    adamLoad[DatabaseVariantAnnotation, U](filePath, predicate, projection)
+    adamLoad[DatabaseVariantAnnotation](filePath, predicate, projection)
   }
 
-  def loadVariantAnnotations[U <: ADAMPredicate[DatabaseVariantAnnotation]](
+  def loadVariantAnnotations(
     filePath: String,
-    predicate: Option[Class[U]] = None,
+    predicate: Option[FilterPredicate] = None,
```

Comment:
Just to clarify. You want the `predicate` argument removed? If so, how will the predicate be available to the final `loadParquetVarientAnnotations` which can use predicate pushdown? Does it make more sense to fast fail here when there is `Some(predicate)` but the format is `vcf` instead of Parquet?